File: Benchmarking/src/gen_5_29.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [[Double]]
    var p2: [[Double]]
    var p3: [[Double]]
  }
  @inline(never)
  func f0(_ v0: s0, _ v1: s0, _ v2: [s0], _ v3: [[s0]], _ v4: s0, _ v5: s0, _ v6: s0, _ v7: Double) -> Double {
    let v9: [[Double]] = v5.p0
    var v21: [[Double]] = v9
    let v35: [Double] = v21[0]
    var v28: [Double] = v35
    let v64: Double = v28[0]
    let v68: [[Double]] = v0.p3
    let v30: [Double] = v68[0]
    v21[0] = v30
    v28[0] = v7
    let v81: [Double] = v21[0]
    v28 = v81
    return v64
  }
  func benchmark() {
    let v0: s0 = s0(p0: [[0.0]], p1: [[1.0]], p2: [[2.0]], p3: [[3.0]])
    let v1: s0 = s0(p0: [[4.0]], p1: [[5.0]], p2: [[6.0]], p3: [[7.0]])
    let v2: [s0] = [s0(p0: [[8.0]], p1: [[9.0]], p2: [[10.0]], p3: [[11.0]]), s0(p0: [[12.0]], p1: [[13.0]], p2: [[14.0]], p3: [[15.0]])]
    let v3: [[s0]] = [[s0(p0: [[16.0]], p1: [[17.0]], p2: [[18.0]], p3: [[19.0]])]]
    let v4: s0 = s0(p0: [[20.0]], p1: [[21.0]], p2: [[22.0]], p3: [[23.0]])
    let v5: s0 = s0(p0: [[24.0]], p1: [[25.0]], p2: [[26.0]], p3: [[27.0]])
    let v6: s0 = s0(p0: [[28.0]], p1: [[29.0]], p2: [[30.0]], p3: [[31.0]])
    let v7: Double = 32.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 32.0
    for _ in 1...1000 {
      result = f0(v0, v1, v2, v3, v4, v5, v6, v7)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()

File: Benchmarking/src/gen_5_29.cpp
  #include <vector>
  #include <iostream>
  #include <chrono>
  struct s0 {
    std::vector<std::vector<double>> p0;
    std::vector<std::vector<double>> p1;
    std::vector<std::vector<double>> p2;
    std::vector<std::vector<double>> p3;
    s0(std::vector<std::vector<double>> v0, std::vector<std::vector<double>> v1, std::vector<std::vector<double>> v2, std::vector<std::vector<double>> v3): p0(v0), p1(v1), p2(v2), p3(v3) { }
  };
  __attribute__((noinline))
  double f0(const s0 &v0, const s0 &v1, const std::vector<s0> &v2, const std::vector<std::vector<s0>> &v3, const s0 &v4, const s0 &v5, const s0 &v6, const double &v7) {
    const std::vector<std::vector<double>> v9 = v5.p0;
    std::vector<std::vector<double>> v21 = v9;
    const std::vector<double> v35 = v21[0];
    std::vector<double> v28 = v35;
    const double v64 = v28[0];
    const std::vector<std::vector<double>> v68 = v0.p3;
    const std::vector<double> v30 = v68[0];
    v21[0] = v30;
    v28[0] = v7;
    const std::vector<double> v81 = v21[0];
    v28 = v81;
    return v64;
  }
  int main() {
    s0 v0({ { { 0.0 } }, { { 1.0 } }, { { 2.0 } }, { { 3.0 } } });
    s0 v1({ { { 4.0 } }, { { 5.0 } }, { { 6.0 } }, { { 7.0 } } });
    std::vector<s0> v2({ { { { 8.0 } }, { { 9.0 } }, { { 10.0 } }, { { 11.0 } } }, { { { 12.0 } }, { { 13.0 } }, { { 14.0 } }, { { 15.0 } } } });
    std::vector<std::vector<s0>> v3({ { { { { 16.0 } }, { { 17.0 } }, { { 18.0 } }, { { 19.0 } } } } });
    s0 v4({ { { 20.0 } }, { { 21.0 } }, { { 22.0 } }, { { 23.0 } } });
    s0 v5({ { { 24.0 } }, { { 25.0 } }, { { 26.0 } }, { { 27.0 } } });
    s0 v6({ { { 28.0 } }, { { 29.0 } }, { { 30.0 } }, { { 31.0 } } });
    double v7(32.0);
    auto start = std::chrono::high_resolution_clock::now();
    double result;
    for (int i = 0; i < 1000; i ++) {
      result = f0(v0, v1, v2, v3, v4, v5, v6, v7);
    }
    auto end = std::chrono::high_resolution_clock::now();
    double value = *((double*) &result);
    std::cout << value << "\n";
    std::cout << std::chrono::duration_cast<std::chrono::nanoseconds>(end-start).count();
    std::cout << "\n";
    return 0;
  }
